pkg/testing/define/define_flags.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package define
+
+import (
+	"flag"
+	"strconv"
+	"strings"
+	"testing"
+)
+
+type optionalBoolFlag struct {
+	set   bool
+	value bool
+}
+
+func (o *optionalBoolFlag) String() string {
+	if !o.set {
+		return "<not set>"
+	}
+	return strconv.FormatBool(o.value)
+}
+
+func (o *optionalBoolFlag) Set(s string) error {
+	o.set = true
+	o.value = s == "" || s == "true"
+	return nil
+}
+
+func (o *optionalBoolFlag) IsSet() bool {
+	return o.set
+}
+
+func (o *optionalBoolFlag) Value() bool {
+	return o.value
+}
```

Comment:
Is all this really worth it vs just using a `*bool` and checking for `nil` directly?

Reply:
Simplified it a bit :wink: